modin/pandas/dataframe.py

```diff
@@ -2451,8 +2454,9 @@ def __setattr__(self, key, value):
             Value to set.
         """
         # We have to check for this first because we have to be able to set
-        # _query_compiler before we check if the key is in self
-        if key in ["_query_compiler"] or key in self.__dict__:
+        # _query_compiler and _siblings before we check whether either key is
+        # in self.
+        if key in ["_query_compiler", "_siblings"] or key in self.__dict__:
```

Comment:
How will the `_siblings` assignment take place in this case?

Reply:
In this case, we just set the `_siblings` variable as usual [here](https://github.com/modin-project/modin/blob/47b8a1ce82c7ff6d786da355edb034ffb1c21b4f/modin/pandas/dataframe.py#L2463). We do the same thing with `_query_compiler`. The point of the `pass` is to avoid trying to call `__setitem__` with these Modin instance variables.

Reply:
Can we add a comment explaining this codepath please?

Reply:
@RehanSD my explanation offline last week was incorrect. It turns out that if we remove the check for `_query_compiler` in master, or for either variable in this PR, we get an infinite recursion when we try to construct any dataframe. I added an explanation in a comment.